src/node_crypto.cc

```diff
@@ -4753,12 +4775,29 @@ void ECDH::SetPublicKey(const FunctionCallbackInfo<Value>& args) {
   EC_POINT* pub = ecdh->BufferToPoint(Buffer::Data(args[0].As<Object>()),
                                       Buffer::Length(args[0].As<Object>()));
   if (pub == nullptr)
-    return;
+    return env->ThrowError("Failed to convert Buffer to EC_POINT");
 
   int r = EC_KEY_set_public_key(ecdh->key_, pub);
   EC_POINT_free(pub);
   if (!r)
-    return env->ThrowError("Failed to convert BN to a private key");
+    return env->ThrowError("Failed to set EC_POINT as the public key");
+}
+
+bool ECDH::IsKeyPairValid() {
+  int result = EC_KEY_check_key(key_);
+  if (0 == result) {
+    // Try to discard from the OpenSSL error stack.
+    // EC_KEY_check_key can return zero without pushing to the stack.
+    // Does not hurt to pop something that was accidentally left there
+    // and won't fail if stack is empty.
+    ERR_get_error();
+  }
+  return 1 == result;
+}
```

Comment:
I would suggest using `ERR_set_mark()` and `ERR_pop_to_mark()`.  That will leave the existing error stack intact.